frontend/express/public/javascripts/countly/countly.vue.components.js

```diff
@@ -372,4 +373,2886 @@
+                        undoRow.remove();
+                        self.nLocks--;
+                        triggeringRow.show();
+                        if (callbacks.undo) {
+                            callbacks.undo();
+                        }
+                    });
+                },
+                optionEvent: function(action) {
+                    var self = this,
+                        focusedRef = this.focusedRow;
+
+                    this.$emit(action.event, focusedRef, function(options) {
+                        if (options.undo) {
+                            self.softAction(focusedRef, options.undo.message, {
+                                commit: function() {
+                                    self.$emit(options.undo.commit, focusedRef);
+                                }
+                            });
+                        }
+                    });
+                }
+            },
+            template: '<div class="cly-vue-datatable-w" ref="wrapper">\
+                            <div ref="buttonMenu" class="cly-button-menu" tabindex="1" v-if="hasOptions">\
+                                <a class="item" @click="optionEvent(optionItem.action)" v-for="(optionItem, j) in optionItems" :key="j"><i :class="optionItem.icon"></i><span>{{optionItem.label}}</span></a>\
+                            </div>\
+                            <table ref="dtable" cellpadding="0" cellspacing="0" class="d-table-vue-wrapper"></table>\
+                        </div>'
+        }
+    ));
+
+    Vue.component("cly-tabs", countlyBaseComponent.extend(
+        // @vue/component
+        {
+            mixins: [
+                _mixins.i18n
+            ],
+            props: {
+                value: { default: null, type: String },
+                skin: { default: "main", type: String}
+            },
+            data: function() {
+                return {
+                    tabs: []
+                };
+            },
+            computed: {
+                skinClass: function() {
+                    if (["main", "graphs"].indexOf(this.skin) > -1) {
+                        return "tabs-" + this.skin + "-skin";
+                    }
+                    return "tabs-main-skin";
+                },
+                numberOfTabsClass: function() {
+                    return "tabs-" + this.tabs.length;
+                },
+                activeContentId: function() {
+                    return this.value;
+                }
+            },
+            mounted: function() {
+                this.tabs = this.$children;
+                if (!this.value) {
+                    this.$emit("input", this.tabs[0].tId);
+                }
+            },
+            methods: {
+                setTab: function(tId) {
+                    this.$emit("input", tId);
+                }
+            },
+            template: '<div class="cly-vue-tabs" v-bind:class="[skinClass]">\
```

Comment:
Codacy found an issue: [Bad escaping of EOL. Use option multistr if needed.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=6339769)